### DISCUSSION TITLE
Support popup-style cycle-able diagnostics

### DISCUSSION THREAD

**Discussion by kas2020-commits** [helix-editor/helix#10499] [General]
First off, love both the Helix project and community, been using it as my main editor for over a year now!

A couple notes as primer before my ask:

- #6417 is currently being worked on by the core team, which introduces a new way to see diagnostics.
- #9974 was just recently merged, which implemented popup cycling through a list of results returned by language servers

so given that there's precedence now for a) adding a new way to see diagnostics and b) cycling through LSP results in a single popup window, I was hoping we could combine the 2 ideas and add the option to view diagnostics in a popup window where we can cycle through all the diagnostics at cursor position?

And before it's asked, yes, if the user choses this option (which would not be the default) then diagnostics won't get automatically shown on the screen. Instead, the user would manually have to use `]d`, `[d` or a keymap that would show diagnostics under cursor (is that already implemented? AFAIK the answer's no but I'm not sure). I actually prefer this behavior - I'm not a huge fan of having a bunch of text show up then disappear when I'm quickly navigating my code.

But what do we think about this idea? Given that most IDEs, vscode and neovim all support this feature, I think it's UX lots of devs would be familiar with already. I'd be willing to drive this myself if such a feat would get approved.

**Comment by pascalkuthe** [comment]
I don't want to support additional ways to display diagnostic at some point its just bloat/too much config/too much maintaince requirement. I am very happy with the state inline diagnostics now

**Reply by zetashift** [reply]
Will it be easy to script new ways to view the diagnostic? I understand it brings in code that's not necessarily needed.

But, for example, even tho inline diagnostics help, sometimes I would like to have a full width reading of the diagnostic. The pop-up also covers a good chunk of the viewport so code isn't always visible.
Emacs `vertico` package is a great way to show stuff like this, so if something like that would be scriptable(could also be useful for a quickfix pane), that'd be awesome!

![image](https://github.com/helix-editor/helix/assets/1857826/9e8e6247-53b0-4fda-9233-b1ba58c824f9)

**Reply by kirawi** [reply]
If it helps, you can use `:yank-diagnostic` and paste it somewhere to look at it at the same time.
